### ISSUE TITLE
`//` repeat last search is broken

### ISSUE THREAD

**Issue by keforbes** [vrapper/vrapper#812]
Noticed in #811.

```
java.lang.ArrayIndexOutOfBoundsException: 0
	at net.sourceforge.vrapper.vim.modes.commandline.SearchCommandParser.getKeyWord(SearchCommandParser.java:126)
	at net.sourceforge.vrapper.vim.modes.commandline.SearchMode.doIncSearch(SearchMode.java:139)
```

```
java.lang.ArrayIndexOutOfBoundsException: 0
	at net.sourceforge.vrapper.vim.modes.commandline.SearchCommandParser.parseSearchCommand(SearchCommandParser.java:103)
	at net.sourceforge.vrapper.vim.modes.commandline.SearchCommandParser.parseAndExecute(SearchCommandParser.java:39)
	at net.sourceforge.vrapper.vim.modes.commandline.AbstractCommandParser.parseAndExecute(AbstractCommandParser.java:322)
	at net.sourceforge.vrapper.vim.modes.commandline.AbstractCommandParser.type(AbstractCommandParser.java:174)
```

Looks like our regex is now invalid:
```
split("(?<!\\\\)\\"+first)[0]
```

**Comment by keforbes** [comment]
`/foo` <-- works
`/foo/e+2` <-- works
`//e+2` <-- works
`//` <-- fails

The errors we're hitting are trying to parse any potential `e+2`, `b-3` movements after the search string.  If no extra movements are defined *and* we're just reusing the last search, it fails.  Oddly, enough, if you hit `//` then hit `enter` you'll go to a newline (because the stack trace aborted the `enter` handling).  If you then hit `enter` again (on that newline), the operation actually succeeds. :man_shrugging:

**Comment by albertdev** [comment]
Huh. Now that I see it and read the JavaSE docs I think this must have never worked right. `String.split(str)` is equal to `String.split(str, 0)` and the latter has this [in its documentation](https://docs.oracle.com/javase/10/docs/api/java/lang/String.html#split%28java.lang.String,int%29):

>  If n is zero then the pattern will be applied as many times as possible, the array can have any length, and trailing empty strings will be discarded. 

The regex matches 2 times so it would have 3 candidates in the resulting array: an empty string before both separators, an empty string in between and one after them. However, when it strips all trailing empty strings it would remove all candidates, hence we end up with an empty array and a crash.

Isn't it possible that this bug has simply been lurking in the code for ages? Using the "run search again" functionality is 99% of the time done using `n` or `/<CR>`. The only difference in using `//<CR>` is that it has the slight semantic difference of being a shorter version of `//{offset}<CR>` (in other words: it repeats the last search but clears the offset). That seems such an edge case that I could understand we'd miss it.

The fix seems simple then after reading the documentation: pass a negative value as the second parameter and you will trigger this behavior:

>  If n is non-positive then the pattern will be applied as many times as possible and the array can have any length.

In other words, you should then get 2 or 3 empty strings back. (Note that there's also something in the docs about empty strings and zero-width matches, that's why I'm not 100% sure of the number of items without trying it.)

**Comment by keforbes** [comment]
Thanks @albertdev!  I think you're right, I've always just used `/` to repeat last search and while `//` *should* be functionally equivalent we probably never explicitly coded for it.

Adding a negative value should fix the behavior.  It's fine for the string to be empty, we just need the array index to exist to avoid the `ArrayIndexOutOfBoundsException`.  I'll give this a try next time I get a chance.

**Comment by albertdev** [comment]
I started working on this one, it only needs some unit tests and then it is good to go.
